coderd/metricscache/metricscache.go

```diff
@@ -198,12 +198,18 @@ func (c *Cache) run(ctx context.Context, name string, interval time.Duration, re
 				logger.Error(ctx, "refresh metrics failed", slog.Error(err))
 				continue
 			}
-			logger.Debug(ctx, "metrics refreshed", slog.F("took", time.Since(start)))
+			logger.Debug(ctx, "metrics refreshed", slog.F("took", c.clock.Since(start)))
 			break
 		}
+	}
 
+	doRefresh(ctx, c.clock.Now())
+
+	for {
 		select {
-		case <-ticker.C:
+		case tick := <-ticker.C:
+			ticker.Stop()
+			doRefresh(ctx, tick)
```

Comment:
the `case <-c.done:` arm of this select is unhittable because a waitgroup on `run()` prevents closing the channel until this function exits.  Without that case, this Ticker/select is now the standard form covered by `c.clock.TickerFunc`. This allows you to force a tick just by advancing the clock, without the need for traps on `Reset`.

However, you still need to set a trap on the call to `TickerFunc` itself, lest advancing the clock race with starting the ticker and leading to a flaky test.